packages/gatsby/src/utils/source-nodes.ts

```diff
@@ -68,20 +73,26 @@ function getStaleNodes(state, nodes) {
 /**
  * Find all stale nodes and delete them
  */
-function deleteStaleNodes(state, nodes) {
+function deleteStaleNodes(state: IGatsbyState, nodes: Node[]): void {
   const staleNodes = getStaleNodes(state, nodes)
 
   if (staleNodes.length > 0) {
     staleNodes.forEach(node => deleteNode({ node }))
   }
 }
 
-module.exports = async ({ webhookBody = {}, parentSpan } = {}) => {
+export default async ({
+  webhookBody,
+  parentSpan,
+}: {
+  webhookBody?: unknown
+  parentSpan?: Span
+}): Promise<void> => {
   await apiRunner(`sourceNodes`, {
     traceId: `initial-sourceNodes`,
     waitForCascadingActions: true,
     parentSpan,
-    webhookBody,
+    webhookBody: webhookBody || {},
```

Comment:
I think this changes semantics (tho not sure) - in case of null `webhookBody` this turns it to empty object which means that checks like `if (webhookBody)` will no longer work as they used to

Reply:
We're actually planning to change this to default to undefined, so this can be removed

Reply:
I would say that his might be breaking change - right now there is guarantee that `webhookBody` is not falsy.

Also check on https://github.com/gatsbyjs/gatsby/pull/15564#discussion_r304475346 (comments about setting `{}` default in the PR that introduced it)